module/support/ReactionStats/src/ReactionStats.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ReactionStats.hpp"
+
+#include "extension/Configuration.hpp"
+
+#include "message/support/nuclear/ReactionStatistics.hpp"
+
+namespace module::support {
+
+    using extension::Configuration;
+    using NUClear::message::ReactionStatistics;
+    using ReactionStatisticsProto = message::support::nuclear::ReactionStatistics;
+
+    ReactionStats::ReactionStats(std::unique_ptr<NUClear::Environment> environment) : Reactor(std::move(environment)) {
+
+        on<Configuration>("ReactionStats.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file ReactionStats.yaml
+        });
+
+        on<Trigger<ReactionStatistics>>().then([this](const ReactionStatistics& stats) {
+            auto reactionData               = std::make_unique<ReactionStatisticsProto>();
+            reactionData->name              = stats.identifier[0];
+            reactionData->trigger_name      = stats.identifier[1];
+            reactionData->function_name     = stats.identifier[2];
+            reactionData->reaction_id       = stats.reaction_id;
+            reactionData->task_id           = stats.task_id;
+            reactionData->cause_reaction_id = stats.cause_reaction_id;
+            reactionData->cause_task_id     = stats.cause_task_id;
+            reactionData->emitted           = 0;
+            reactionData->started           = 0;
+            reactionData->finished          = 0;
+            // TODO fix this
+            /*reactionData.emitted           = getUtcTimestamp<std::chrono::microseconds>(stats.emitted);
+            reactionData.started           = getUtcTimestamp<std::chrono::microseconds>(stats.started);
+            reactionData.finished          = getUtcTimestamp<std::chrono::microseconds>(stats.finished);*/
```

Comment:
You should be able to just attach these directly to the message with no function, each of the `stats.x` times are just NUClear::clock::time_point objects.
These can be directly assigned into the protobuf so long as the protobufs type is a timestamp.

So change the protobuf type to `google.protobuf.Timestamp` and Neutron will figure out the rest.

Reply:
Sweet, I couldn't figure it out and decided it was a future me problem as it wasn't a priority 😆 I'll chuck it in, thanks 🙂